### DISCUSSION TITLE
Adding a Custom Message Box Below External Grader Results in PrairieLearn

### DISCUSSION THREAD

**Discussion by aalrabah** [PrairieLearn/PrairieLearn#10225] [Dev]
Hello PrairieLearn community,

I'm working on customizing the feedback for externally graded questions in PrairieLearn. Currently, when a student submits an answer, they see a message box containing grading results and test cases. This box is generated by `<pl-external-grader-results>.` from `question.html`

What I'd like to achieve is to add a second, custom message box below the existing one. This new box would contain additional feedback that I can control directly from the grading script.

My question is: How can I implement this additional message box? Should I approach this through the `serverFilesCourse`, the `question.html` or is there another recommended method?

**Comment by jonatanschroeder** [comment]
Note that you can set `message` and `output` in results.json, which will show on top of the tests. Something like:

```
{
  "message": "...",
  "output": "...",
  "tests: [
    ...
  ],
  ...
}
```

Is this what you're looking for? If not, can you be a bit more specific? There are alternatives you can add, but before describing them it would be useful to get a bit more information.

**Reply by aalrabah** [reply]
If you see the attached image, there is a feedback box with a black background that appears right below the "Message" section whenever an answer is submitted. I would like to create an additional message box right below the existing feedback box, so I can have two separate feedback boxes.

<img width="579" alt="Screenshot 2024-07-10 at 3 06 03 PM" src="https://github.com/PrairieLearn/PrairieLearn/assets/94386936/2117d3ed-6625-4cd2-8ed8-368e8b544935">

**Reply by jonatanschroeder** [reply]
Have you tried output in addition to message?

**Reply by aalrabah** [reply]
Thank you very much got it!
